omega/linear_regression/LinearRegressionClassifier.py

```diff
@@ -0,0 +1,81 @@
+import torch as ch
+import logging
+from torch.autograd import Variable
+
+
+logging.basicConfig()
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+class LinearRegressionClassifier(object):
+    def __init__(self, x, y):
+        self.x = x
+        self.y = y
+
+    def fit(self, option):
+        self.option = option
+        ch.manual_seed(2)
+        if option == 'Analytical':
+            self.cov_x_y = []
+            self.var_x = []
+            mean_x = ch.mean(self.x)
+            mean_y = ch.mean(self.y)
+            for i in range(len(self.x)):
+                self.cov_x_y.append((self.x[i] - mean_x) * (self.y[i] - mean_y))
+                self.var_x.append((self.x[i] - mean_x) ** 2)
+            self.b1 = sum(self.cov_x_y) / sum(self.var_x)
+            self.b0 = mean_y - self.b1 * mean_x
+        if option == 'SGD':
+            x_data = Variable(ch.Tensor(self.x), requires_grad=False)
+            y_data = Variable(ch.Tensor(self.y), requires_grad=False)
+            self.beta = Variable(ch.randn(1, 1), requires_grad=True)
+            self.alpha = Variable(ch.randn(1), requires_grad=True)
+            optimizer = ch.optim.SGD([self.beta, self.alpha], lr=0.01)
```

Comment:
We need to write our own optimizers and limit the use of `pytorch` for "numerical operations" only. We need to define a helper function for these optimizers.